src/puppetlabs/puppetdb/cli/export.clj

```diff
@@ -46,213 +52,107 @@
-  [node :- s/Str
-   base-url :- utils/base-url-schema]
-  (let [base-url (merge {:version api-version} base-url)]
-    (when-let [body (parse-response
-                     (client/get
-                      (str (utils/base-url->str base-url)
-                           "/factsets?query="
-                           (url-encode
-                            (format "[\"=\",\"certname\",\"%s\"]" node)))
-                      {:accept :json}))]
-      (map
-       factsets-expansion
-       body
-       (repeat base-url)))))
+(pls/defn-validated facts->tar :- utils/tar-item
+  "Creates a tar-item map for the collection of facts"
+  [{:keys [certname] :as facts} :- {s/Keyword s/Any}]
+  {:msg (format "Writing facts for node '%s'" certname)
+   :file-suffix ["facts" (format "%s.json" certname)]
+   :contents (json/generate-pretty-string facts)})
 
 (pls/defn-validated facts-for-node :- (s/maybe (s/pred seq? 'seq?))
   "Retrieves the factset for a given certname, returning a compatible
   wire format."
-  [base-url :- utils/base-url-schema
+  [query-fn
+   completion-fn
    node :- s/Str]
-  (-> (facts-for-node-query node base-url)
-      factsets/factsets-query->wire-v4))
-
-(pls/defn-validated facts->tar :- utils/tar-item
-  "Creates a tar-item map for the collection of facts"
-  [node :- String
-   facts :- {s/Keyword s/Any}]
-  {:msg (format "Writing facts for node '%s'" node)
-   :file-suffix ["facts" (format "%s.json" node)]
-   :contents (json/generate-pretty-string facts)})
+  (->> (query-fn "factsets" node)
+       (complete-unexpanded-fields completion-fn [:facts])
+       factsets/factsets-query->wire-v4
+       (map facts->tar)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Report Exporting
 
-(pls/defn-validated reports-expansion :- {s/Keyword s/Any}
-  [record :- {s/Keyword s/Any}
-   base-url :- utils/base-url-schema]
-  (kitchensink/mapvals #(complete-unexpanded % base-url) [:resource_events :metrics :logs] record))
-
-(pls/defn-validated reports-for-node-query :- (s/maybe (s/pred seq? 'seq?))
-  "Given a node name, retrieves the reports for the node."
-  [node :- s/Str
-   base-url :- utils/base-url-schema]
-  (let [base-url (merge {:version api-version} base-url)]
-    (when-let [body (parse-response
-                     (client/get
-                      (str (utils/base-url->str base-url)
-                           "/reports?query="
-                           (url-encode (format "[\"=\",\"certname\",\"%s\"]"
-                                               node)))
-                      {:accept :json}))]
-      (map #(reports-expansion % base-url) body))))
+(pls/defn-validated report->tar :- utils/tar-item
+  "Create a tar-item map for the `report`"
+  [{:keys [certname configuration_version start_time] :as report}]
+  (let [unique-seed (str (json/parse-string
+                          (json/generate-pretty-string start_time)) configuration_version)
```

Comment:
That was the issue I was having yesterday with the anonymizer hash being different, basically `start_time` is a timestamp and `generate-pretty-string` formats it to a string (like we used to with the old import querying), I decided to do this so that the string basically follows the same codepath it used to.